mqtt_client.c

```diff
@@ -208,6 +209,10 @@ static esp_err_t esp_mqtt_set_ssl_transport_properties(esp_transport_list_handle
                      goto esp_mqtt_set_transport_failed);
     }
 
+    if (cfg->crt_bundle_attach != NULL) {
+        esp_transport_ssl_crt_bundle_attach(ssl, cfg->crt_bundle_attach);
```

Comment:
This should be `#ifdef`'ed out for earlier versions of IDF, as we typically backport the same version of the `esp-mqtt` to all supported releases of IDF.

You can setup a macro in the [`supported-features.h`](https://github.com/espressif/esp-mqtt/blob/master/include/mqtt_supported_features.h#L51-L54) header. Note that the cert-bundle was introduced in IDF `v4.2`.

Reply:
Hi @david-cermak,
for me it would be important to have this feature. It doesn't matter if it is taken from my commit or if you want to address it internally. 
Which one would be result in a faster integration within the stable version? ;)

Thank you for your code suggestions. I would implement them anyway and create a new pull request.

Reply:
@david-cermak changes are in the new commit. thanks for introducing the idf-version defines to me!